File: src/main/java/net/ndrei/teslacorelib/gui/MachineNameGuiPiece.java
package net.ndrei.teslacorelib.gui;

import net.minecraft.client.renderer.GlStateManager;
import net.minecraft.client.resources.I18n;

/**
 * Created by CF on 2016-12-30.
 */
public class MachineNameGuiPiece extends BasicContainerGuiPiece {
    private String unlocalizedName;

    public MachineNameGuiPiece(String unlocalizedName, int left, int top, int width, int height) {
        super(left, top, width, height);

        this.unlocalizedName = unlocalizedName;
    }

    @Override
    public void drawBackgroundLayer(BasicTeslaGuiContainer container, int guiX, int guiY, float partialTicks, int mouseX, int mouseY) {
        if ((this.unlocalizedName != null) && (this.unlocalizedName.length() > 0)) {
            String title = I18n.format(this.unlocalizedName);
            container.mc.fontRendererObj.drawString(title, guiX + this.getLeft(), guiY + this.getTop(), 4210751);
            GlStateManager.color(1.0f, 1.0f, 1.0f, 1.0f);
        }
    }
}

File: src/main/kotlin/net/ndrei/teslacorelib/gui/MachineNameGuiPiece.kt
package net.ndrei.teslacorelib.gui

import net.minecraft.client.renderer.GlStateManager
import net.minecraft.client.resources.I18n
import net.ndrei.teslacorelib.compatibility.FontRendererUtil

/**
 * Created by CF on 2017-06-28.
 */
class MachineNameGuiPiece(private val unlocalizedName: String?, left: Int, top: Int, width: Int, height: Int)
    : BasicContainerGuiPiece(left, top, width, height) {

    override fun drawBackgroundLayer(container: BasicTeslaGuiContainer<*>, guiX: Int, guiY: Int, partialTicks: Float, mouseX: Int, mouseY: Int) {
        if ((this.unlocalizedName != null) && this.unlocalizedName.isNotEmpty()) {
            val title = I18n.format(this.unlocalizedName)
            FontRendererUtil.fontRenderer.drawString(title, guiX + this.left, guiY + this.top, 4210751)
            GlStateManager.color(1.0f, 1.0f, 1.0f, 1.0f)
        }
    }
}
